Projects/Social-Trading/SA/Function-Libraries/SocialEntitiesProfile.js

```diff
@@ -473,6 +478,48 @@ exports.newSocialTradingFunctionLibrariesSocialEntitiesProfile = function () {
 
                 SA.projects.network.globals.memory.maps.USER_PROFILES_BY_ID.set(userProfile.id, inMemoryUserProfile)
             }
+            
+            function saveApiAppFile() {
+             
+                const SOCIAL_TRADING_REPO_NAME = profileMessage.socialEntityHandle + "-" + profileMessage.socialEntityType.replace(' ', '-') + "-Data"
+                /*
+                Try to load the Secrets file. 
+                */
+                let filePath = global.env.PATH_TO_SECRETS + '/'
+                let fileName = "ApisSecrets.json"
+                let fileContent 
+                try {
+                    fileContent = SA.nodeModules.fs.readFileSync(filePath + '/' + fileName)
+                }catch(err) {
+                    /* If the file does not exist, then we'll get here.*/
+                }
+                
+                let secretsFile
+                if (fileContent === undefined) {
+                    secretsFile = {
+                        secrets: []
+                    }
+                } else {
+                    secretsFile = JSON.parse(fileContent)
+                }
+
```

Comment:
Might want to validate the token before assigning.

if (profileMessage.storageProviderToken !== undefined) {
    let secret = {
                    nodeCodeName: SOCIAL_TRADING_REPO_NAME,
                    token: profileMessage.storageProviderToken
                }
                secretsFile.secrets.push(secret)
                 /*
                Save Signing Accounts Secrets File
                */
                SA.projects.foundations.utilities.filesAndDirectories.createNewDir(filePath)
                SA.nodeModules.fs.writeFileSync(filePath + '/' + fileName, JSON.stringify(secretsFile, undefined, 4))
} else {
    response = {
result: 'Error',
message: '[Error] profileMessage.storageProviderToken is undefined.'
}
resolve(response);
}

Reply:
you can't submit the form without a storageProviderToken. the pushUserProfileAndPullRequest function will give an error if the storageProviderToken is not valid